src/CoreWCF.NetTcp/src/CoreWCF/Channels/Framing/FramingConnection.cs

```diff
@@ -210,6 +200,57 @@ public async Task SendFaultAsync(string faultString, TimeSpan sendTimeout, int m
             }
         }
 
+        /// <summary>
+        /// Tries to extract the remote endpoint from the given <see cref="ConnectionContext"/> in a
+        /// version agnostic way.
+        /// </summary>
+        /// <param name="context">The ASP.net core connection context to extract the remote endpoint from.</param>
+        /// <returns>The endpoint of the remote party or null if it was not provided.</returns>
+        private static IPEndPoint GetRemoteEndPoint(ConnectionContext context)
+        {
+            // 1st chance: Server might provide remote endpoint via HTTP feature
+            // (mostly the case in ASP.net core v2.x)
+            IHttpConnectionFeature connectionFeature = context.Features.Get<IHttpConnectionFeature>();
+            if (connectionFeature != null)
+            {
+                return new IPEndPoint(connectionFeature.RemoteIpAddress, connectionFeature.RemotePort);
+            }
+
+            // 2nd chance: on ASP.net core 5.0 the ConnectionContext has a direct Property RemoteEndpoint
+            // via baseclass.
+            var net5RemoteEndPointPropertyAccessor = s_net5RemoteEndPointPropertyAccessor;
+            if (net5RemoteEndPointPropertyAccessor != null)
+            {
+                return net5RemoteEndPointPropertyAccessor(context);
+            }
+
+            // last chance: server does likely not support access to remote endpoint. could be
+            // a non-tcp server like the ASP.net core test server
+            return null;
+        }
+
+
+        private static readonly Func<ConnectionContext, IPEndPoint> s_net5RemoteEndPointPropertyAccessor =
+            BuildNet5RemoteEndPointPropertyAccessor();
+
+        private static Func<ConnectionContext, IPEndPoint> BuildNet5RemoteEndPointPropertyAccessor()
+        {
+            // https://github.com/dotnet/aspnetcore/blob/v5.0.9/src/Servers/Connections.Abstractions/src/BaseConnectionContext.cs
+            var property =
+                typeof(ConnectionContext).GetProperty("RemoteEndPoint", BindingFlags.Instance | BindingFlags.Public);
+            if (property == null)
+            {
+                return null;
+            }
+
+            // context => context.RemoteEndPoint as IPEndpoint
+            var contextParam = Expression.Parameter(typeof(ConnectionContext), "context");
```

Comment:
FYI, property.GetGetMethod returns a MethodInfo and you can use Delegate.CreateDelegate to create an open delegate for getting the property value. I'm fine with your approach too though. An interesting thing is your general approach used for fields is significantly faster than using reflection as the methods to access fields result in around 70 lines of native code running in the clr when using reflection, but expressions compile to IL and very directly access the field in memory. Expressions are also slower in AOT scenarios (at least they used to be for UWP) as they are interpreted at runtime and just end up using the reflection method in the end. I don't think that's a concern here though.

Reply:
I used quite some expression compiling in the past due to this benefit of runtime performance. Of course it comes at some initial cost for compiling the lamdba. I think this dynamically emitted lambdas even have a chance to get inlined by the runtime which makes them almost as fast as if there would be a normal c# access. 

I would say also for AOT cases where the expression tree is still interpreted instead of emitted to IL, the complexity of the tree is simple enough to not cause real troubles.